tests/foreman/ui_airgun/test_domain.py

```diff
@@ -0,0 +1,249 @@
+# -*- encoding: utf-8 -*-
+"""Test class for Domain UI
+
+:Requirement: Domain
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: UI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from fauxfactory import (
+    gen_string,
+    gen_utf8
+)
+import pytest
+
+from robottelo.constants import DOMAIN
+from robottelo.datafactory import parametrized
+from robottelo.decorators import (
+    parametrize,
+    tier2,
+    upgrade
+)
+
+
+def valid_domain_names(length):
```

Comment:
Please move this and the next func to robottelo/datafactory.py (take a look at valid_data_list for the reference).
Also i'd make `length` optional to make it consistent with other datafactory generators - if length is not passed then use random one.